src/Database/Redis/ProtocolPipelining.hs

```diff
@@ -42,14 +43,15 @@ data Connection = Conn
     -- ^ Number of pending replies and thus the difference length between
     --   'connReplies' and 'connPending'.
     --   length connPending  - pendingCount = length connReplies
+  , hooks         :: Hooks
   }
 
 
 fromCtx :: CC.ConnectionContext -> IO Connection
-fromCtx ctx = Conn ctx <$> newIORef [] <*> newIORef [] <*> newIORef 0
+fromCtx ctx = Conn ctx <$> newIORef [] <*> newIORef [] <*> newIORef 0 <*> pure defaultHooks
 
-connect :: NS.HostName -> CC.PortID -> Maybe Int -> IO Connection
-connect hostName portId timeoutOpt = do
+connect :: NS.HostName -> CC.PortID -> Maybe Int -> Hooks -> IO Connection
+connect hostName portId timeoutOpt hooks = do
```

Comment:
Since Hooks is a breaking change. Would it make more sense to expose a new function `connectWithHooks` and the current `connect` function would internally use `defaultHooks` so the signature doesn't change.

Reply:
Sure!